statisticservice/statistics-service.js

```diff
@@ -19,6 +19,149 @@ if (mongoose.connection.readyState === 0) {
     .catch(err => console.error('MongoDB connection error:', err));
 }
 
+function isValidDateString(dateStr) {
+  // Ensure it matches YYYY-MM-DD
+  if (!/^\d{4}-\d{2}-\d{2}$/.test(dateStr)) return false;
+
+  const date = new Date(dateStr);
+  if (isNaN(date.getTime())) return false;
+
+  const [year, month, day] = dateStr.split('-').map(Number);
+  return date.getUTCFullYear() === year &&
+    date.getUTCMonth() + 1 === month &&
+    date.getUTCDate() === day;
+}
+
+// Get statistics for all users (ordered)
+app.get('/statistics', async (req, res) => {
+
+  // Extract query parameters for filtering and sorting
+  const {
+    sort = 'gamesPlayed',
+    order = 'desc',
+    limit = 50,
+    offset = 0,
+    gameType,
+    minGames,
+    minScore,
+    registeredBefore,
+    registeredAfter
+  } = req.query;
+
+  // Validate sort field
+  const validSortFields = ['username', 'gamesPlayed', 'totalVisits', 'registrationDate', 'maxScore'];
```

Comment:
When I try to sort the games by `totalScore` the app is responding with a `400` status code. I think it should be included in these fields and manage it correctly.
```suggestion
  const validSortFields = ['username', 'gamesPlayed', 'totalScore', 'totalVisits', 'registrationDate', 'maxScore'];
```
Also, I see that it allows for filtering by `totalVisits` while in the application I'm not sure if I can order it by that field. Maybe this something not related with the statistics page.

Reply:
Yes, I allowed to filter by more fields than are actually are in the table because I was thinking on improving the view in the future and add more fields to the table explained in a new issue. I just wanted to merge a simple and functional version into the master branch because I did some improvements in the endpoints, so that all the group members could merge those changes into their branches and have a better foundation to develop their work. Perhaps it wasn't a good decision.

Reply:
Oh, okay fine.